app/frontend/src/components/forms/FormSubmission.vue

```diff
@@ -87,16 +110,27 @@ export default {
       loading: true,
     };
   },
-  computed: mapGetters('form', ['form', 'formSubmission']),
+  computed: mapGetters('form', ['form', 'formSubmission', 'permissions']),
   methods: {
-    ...mapActions('form', ['fetchSubmission']),
+    ...mapActions('form', ['fetchSubmission', 'getFormPermissionsForUser']),
     refreshNotes() {
       this.$refs.notesPanel.getNotes();
     },
+
+    canViewAllSubmissions() {
+      const perms = [
+        FormPermissions.SUBMISSION_READ,
+        FormPermissions.SUBMISSION_UPDATE,
+      ];
+      return this.permissions.some((p) => perms.includes(p));
+    },
   },
   async mounted() {
     await this.fetchSubmission({ submissionId: this.submissionId });
     this.loading = false;
+
+    // get current user's permissions on associated form
+    await this.getFormPermissionsForUser(this.form.id);
```

Comment:
If you're checking for permissions, this MUST be done before loading is set to false. Loading should ONLY flip to false when all async XHR requests have completed.